docs/reference/api/docker_remote_api_v1.21.md

```diff
@@ -2244,6 +2244,127 @@ Status Codes:
 -   **404** – no such exec instance
 -   **500** - server error
 
+## 2.4 Volumes
+
+### List volumes
+
+`GET /volumes`
+
+**Example request**:
+
+  GET /volumes HTTP/1.1
+
+**Example response**:
+
+  HTTP/1.1 200 OK
+  Content-Type: application/json
+
+  {
+    "Volumes": [
+      {
+        "Name": "tardis",
+        "Driver": "local",
+        "Mountpoint": "/var/lib/docker/volumes/tardis"
+      }
+    ]
+  }
+
+Query Parameters:
+
+- **filter** - JSON encoded value of the filters (a map[string][]string) to process on the volumes list. Available filters:
+  -   `dangling=true`
+
+Status Codes:
+
+-   **200** - no error
+-   **500** - server error
+
+### Create a volume
+
+`POST /volumes`
+
+Create a volume
+
+**Example request**:
+
+  POST /volumes HTTP/1.1
+  Content-Type: application/json
+
+  {
+    "Name": "tardis"
+  }
+
+**Example response**:
+
+  HTTP/1.1 201 Created
+  Content-Type: application/json
+
+  {
+    "Name": "tardis"
+    "Driver": "local",
+    "Mountpoint": "/var/lib/docker/volumes/tardis"
+  }
+
+Status Codes:
+
+- **201** - no error
+- **500**  - server error
+
+Json Parameters:
+
+- **Name** - The name for the created volume. If not specified a name is
+    generated
+- **Driver** - Name of the volume driver to use. Default is `local`
+- **DriverOpts** - A mapping of driver options and values. These options are
+    passed directly to the driver, and are driver specific.
+
+### Inspect a volume
+
+`GET /volumes/(name)`
+
+Return low-level informtion on the volume `name`
+
+**Example request**:
+
+    GET /volumes/tardis
+
+**Example resppnse**:
+
+  HTTP/1.1 200 OK
+  Content-Type: application/json
+
+  {
+    "Name": "tardis",
+    "Driver": "local",
+    "Mountpoint": "/var/lib/docker/volumes/tardis"
```

Comment:
Nope, this is not stored (no volume info is stored locally).
At some point we may query the driver about it, but not for now.